pkg/util/mathutil/math.go

```diff
@@ -110,3 +110,31 @@ func Divide2Batches(total, batches int) []int {
 	}
 	return result
 }
+
+// Divide2Batches divies 'items' into 'batches', and returns the items in each batch.
+// The size of each batch is determined by Divide2BatchSizes.
+func Divide2Batches[T any](items []T, batches int, minBatchCnt, maxBatchCnt int) [][]T {
```

Comment:
please move to `external` pkg, it's our algorithm to generate sizes of subtasks for global sort, not that general